pkg/controller/certificates/signer/signer.go

```diff
@@ -185,15 +189,12 @@ func (s *signer) handle(csr *capi.CertificateSigningRequest) error {
 	return nil
 }
 
-func (s *signer) sign(x509cr *x509.CertificateRequest, usages []capi.KeyUsage) ([]byte, error) {
+func (s *signer) sign(x509cr *x509.CertificateRequest, usages []capi.KeyUsage, durationHint *metav1.Duration) ([]byte, error) {
 	currCA, err := s.caProvider.currentCA()
 	if err != nil {
 		return nil, err
 	}
-	der, err := currCA.Sign(x509cr.Raw, authority.PermissiveSigningPolicy{
-		TTL:    s.certTTL,
-		Usages: usages,
-	})
+	der, err := currCA.Sign(x509cr.Raw, s.policy, usages, durationHint)
```

Comment:
`authority.PermissiveSigningPolicy` is intended to be used as named arguments to Sign. Can you preserve that API?

Reply:
It is a container of arguments so that the call site is clear and arguments can be added without breaking API compatibility. Can we keep the package API as is?